src/sage/interfaces/expect.py

```diff
@@ -561,7 +562,7 @@ def _close(self, force=True):
         try:
             if self._expect is not None:
                 self._expect.close(force=force)
-        except ExceptionPexpect:
+        except (PtyProcessError, ExceptionPexpect):
```

Comment:
```suggestion
        except (ExceptionPexpect, OSError):
```

Can you try fixing it here instead? At least here it is reasonably documented what is the reason close may fail.

It doesn't seem we need to catch `PtyProcessError` as it is translated to `ExceptionPexpect` by pexpect itself.

Presumably, nothing needs to be changed in `src/sage/interfaces/sagespawn.pyx`.

Reply:
OK, this seems to work, too. I'll change the branch.